### PR TITLE
Fix issue's title is not updated on dashboard

### LINKED ISSUE #26868: Issue's title is not updated on dashboard
### Description

I have create an issue, than I`ve updated its title
image

But on (my) dashboard it has old name
image

Should issue title be updated across UI ?

### Screenshots

none

### Gitea Version

1.21.0+dev-688-gf01bed244

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

_No response_

### Browser Version

116

### PR DESCRIPTION
**Pull request by CaiCandong**
I checked `templates/user/dashboard/feeds.tmpl` and found two places are out of sync :
- `Opened issue` Action, When you change the issue title, the title in the dashboard is not updated
- `CreatePullRequest` Action ,  When you change the PR title, the title in the dashboard  is not updated

Fix #26868

- Fix issue's title is not updated on the dashboard
- Refactor `templates/user/dashboard/feeds.tmpl`  to remove magic numbers

### CODE REVIEW

File: templates/user/dashboard/feeds.tmpl [hunk 1]
```diff
@@ -95,21 +95,21 @@
...
-					<span class="text truncate issue title">{{index .GetIssueInfos 1 | RenderEmoji $.Context | RenderCodeBlock}}</span>
-				{{else if or (eq .GetOpType 10) (eq .GetOpType 21) (eq .GetOpType 22) (eq .GetOpType 23)}}
+				{{else if .GetOpType.IsActionCreateIssue}}
+					<span class="text truncate issue title">{{.GetIssueTitle | RenderEmoji $.Context | RenderCodeBlock}}</span>
+				{{else if .GetOpType.IsActionCreatePullRequest}}
+					<span class="text truncate issue title">{{.GetIssueTitle | RenderEmoji $.Context | RenderCodeBlock}}</span>
+				{{else if or .GetOpType.IsActionCommentIssue .GetOpType.IsActionApprovePullRequest .GetOpType.IsActionRejectPullRequest .GetOpType.IsActionCommentPull}}
 					<a href="{{.GetCommentLink}}" class="text truncate issue title">{{.GetIssueTitle | RenderEmoji $.Context | RenderCodeBlock}}</a>
-					{{$comment := index .GetIssueInfos 1}}
+					{{$comment := .GetCommentContent}}
```
Comment (CaiCandong) [inline templates/user/dashboard/feeds.tmpl:104]: https://github.com/go-gitea/gitea/assets/50507092/2589c17d-ba2c-4d0c-98d4-bde29067a8f0

File: templates/user/dashboard/feeds.tmpl [hunk 2]
```diff
@@ -95,21 +95,21 @@
...
 					{{end}}
-				{{else if eq .GetOpType 6}}
-					<span class="text truncate issue title">{{index .GetIssueInfos 1 | RenderEmoji $.Context | RenderCodeBlock}}</span>
-				{{else if eq .GetOpType 7}}
-					<span class="text truncate issue title">{{index .GetIssueInfos 1 | RenderEmoji $.Context | RenderCodeBlock}}</span>
-				{{else if or (eq .GetOpType 10) (eq .GetOpType 21) (eq .GetOpType 22) (eq .GetOpType 23)}}
+				{{else if .GetOpType.IsActionCreateIssue}}
+					<span class="text truncate issue title">{{.GetIssueTitle | RenderEmoji $.Context | RenderCodeBlock}}</span>
+				{{else if .GetOpType.IsActionCreatePullRequest}}
+					<span class="text truncate issue title">{{.GetIssueTitle | RenderEmoji $.Context | RenderCodeBlock}}</span>
```
Comment (CaiCandong) [inline templates/user/dashboard/feeds.tmpl:101]: https://github.com/go-gitea/gitea/assets/50507092/e3ffe563-0f1b-475d-8814-3534bfd0700d

File: templates/user/dashboard/feeds.tmpl [hunk 3]
```diff
@@ -95,21 +95,21 @@
 					{{if and (gt $push.Len 1) $push.CompareURL}}
 						<a href="{{AppSubUrl}}/{{$push.CompareURL}}">{{$.locale.Tr "action.compare_commits" $push.Len}} »</a>
 					{{end}}
-				{{else if eq .GetOpType 6}}
-					<span class="text truncate issue title">{{index .GetIssueInfos 1 | RenderEmoji $.Context | RenderCodeBlock}}</span>
-				{{else if eq .GetOpType 7}}
-					<span class="text truncate issue title">{{index .GetIssueInfos 1 | RenderEmoji $.Context | RenderCodeBlock}}</span>
-				{{else if or (eq .GetOpType 10) (eq .GetOpType 21) (eq .GetOpType 22) (eq .GetOpType 23)}}
+				{{else if .GetOpType.IsActionCreateIssue}}
+					<span class="text truncate issue title">{{.GetIssueTitle | RenderEmoji $.Context | RenderCodeBlock}}</span>
```
Comment (CaiCandong) [inline templates/user/dashboard/feeds.tmpl:99]: https://github.com/go-gitea/gitea/assets/50507092/80b0059d-3f0f-4256-8a8d-0685efd36369

**Comment by KN4CK3R** [comment]
If we begin to query live data for the action feed, we may only store the reference id in the action table without other infos.

**Comment by silverwind** [comment]
I guess it depends how this "feed" is seen. Either as an append-only log (like now) or a live, updated view.

Are there other fields besides issue title in that log which can invalidate it? If it's this single event type, we may get away by adding a "renamed issue/pr" event, but it will spam the "log".

**Comment by CaiCandong** [comment]
> If we begin to query live data for the action feed, we may only store the reference id in the action table without other infos.

The existing code has been doing that.
https://github.com/go-gitea/gitea/blob/1859c5b636637d0a827c4b67ab0f59fe4b635f87/templates/user/dashboard/feeds.tmpl#L103

**Comment by CaiCandong** [comment]
> I guess I prefer a mixed approach. append only where we know the event can or should not change anymore (git push), and live data for stuff like issue/pr comments/creation.

After this PR, it seems that there is only one place where the info is not changing anymore : 
https://github.com/go-gitea/gitea/blob/1859c5b636637d0a827c4b67ab0f59fe4b635f87/templates/user/dashboard/feeds.tmpl#L78

**Comment by KN4CK3R** [comment]
> > If we begin to query live data for the action feed, we may only store the reference id in the action table without other infos.
> 
> The existing code has been doing that.

I know, that's not a bad thing about this PR. Github does that too. Could save some space if we don't have to store the issue title anymore for example.

**Comment by CaiCandong** [comment]
> I know, that's not a bad thing about this PR. Github does that too. Could save some space if we don't have to store the issue title anymore for example.

As you can see in the image, `action.Content` holds not only `issue.Title`, but also a few other things.
image

**Comment by KN4CK3R** [comment]
Sure, but if query live data we only need the issue id.

**Comment by CaiCandong** [comment]
> Sure, but if query live data we only need the issue id.

Right, I think `issue` should be rescued from content.

**Pull request closed without merging**